speechbrain/lobes/models/transformer/TransformerASR.py

```diff
@@ -333,11 +454,95 @@ def encode(self, src, wav_len=None, pad_idx=0):
 
         encoder_out, _ = self.encoder(
             src=src,
+            src_mask=src_mask,
             src_key_padding_mask=src_key_padding_mask,
             pos_embs=pos_embs_source,
+            dct_config=dct_config,
         )
+
         return encoder_out
 
+    def encode_streaming(self, src, context: TransformerASRStreamingContext):
+        """
+        Streaming encoder forward pass
+
+        Arguments
+        ---------
+        src : torch.Tensor
+            The sequence (chunk) to the encoder.
+
+        context : TransformerASRStreamingContext
+            Mutable reference to the streaming context. This holds the state
+            needed to persist across chunk inferences and can be built using
+            `make_streaming_context`. This will get mutated by this function.
+
+        Returns
+        -------
```

Comment:
Can we have an example? Or too complicated maybe.

Reply:
Added a docstring example